### ISSUE TITLE
Support for TAB in repl

### ISSUE THREAD

**Issue by avive** [spacemeshos/smrepl#34]
Make TAB key behave like SPACE key in repl keyboard input so the auto-suggestor for the current level will be displayed.

**Comment by Soptq** [comment]
There is a corner case I think we are missing to discuss here: when we press tab for a auto-completion, for example when we press tab with a input of 'wall', go-prompt will automatically insert 'et' to the end of our input. At this moment 'tab' will be treated as "withdraw" or "switch" by go-prompt even though the auto-suggestor is closed, which will cause conflicts with our original intention.

**Comment by avive** [comment]
Good point, I think needs to be handled in the following way: 
1. If tab is pressed after some text is entered which is not a valid command at the current completion level then it should work like it does now - open the suggestor floating area with valid suggestion based on user text entry prefix.
2. If tab is pressed just after a valid command was entered by the user (by typing or by completing from a suggestion) then treat it should be treated as space and open the suggestor floating area for the next level of suggestions.

**Comment by Soptq** [comment]
Yeah, but in order to achieve this, perhaps we need to maintain a fork of `go-prompt` since to my best knowledge there is no such a callback in `go-prompt` that capable of retrieving variables we need here. Nevertheless, I will do more research of it.

**Comment by avive** [comment]
I see. Maybe there are better prompt libs out there that we can use?
